addons/l10n_sg/models/res_bank.py

```diff
@@ -0,0 +1,58 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ResPartnerBank(models.Model):
+    _inherit = 'res.partner.bank'
+
+    proxy_type = fields.Selection(selection_add=[('mobile', 'Mobile Number'), ('uen', 'UEN')],
+                                  ondelete={'mobile': 'set default', 'uen': 'set default'})
+
+    @api.constrains('proxy_type', 'proxy_value', 'partner_id')
+    def _check_sg_proxy(self):
+        for bank in self.filtered(lambda b: b.country_code == 'SG'):
+            if bank.proxy_type not in ['mobile', 'uen', 'none', False]:
+                raise ValidationError(_("The PayNow Type must be either Mobile or UEN to generate a PayNow QR code."))
+
+    @api.depends('country_code')
+    def _compute_display_qr_setting(self):
+        bank_sg = self.filtered(lambda b: b.country_code == 'SG')
+        bank_sg.display_qr_setting = True
```

Comment:
To check, but wouldn't it make sense to only display these settings if the "Display QR-code on invoices" setting is set on the company?